cmd/orphan-finder/main.go

```diff
@@ -67,83 +98,142 @@ var (
 
 var backdateDuration time.Duration
 
-func checkDER(sai certificateStorage, der []byte) (*x509.Certificate, error) {
+// orphanTypeForCert returns precertOrphan if the certificate has the RFC 6962
+// CT poison extension, or certOrphan if it does not. If the certificate is nil
+// unknownOrphan is returned.
+func orphanTypeForCert(cert *x509.Certificate) orphanType {
+	if cert == nil {
+		return unknownOrphan
+	}
+	// RFC 6962 Section 3.1 - https://tools.ietf.org/html/rfc6962#section-3.1
+	poisonExt := asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 11129, 2, 4, 3}
```

Comment:
We have this OID defined in a few places now:
* here
* in test/integration/common_test.go
* in publisher/publisher_test.go
* in ca/ca_test.go

Maybe we should find a place to export this OID from that all of the above places can reference? I couldn't think of an obvious choice during this PR. Something under `test` makes sense except for this tool, where it feels odd to import a test package for a const OID.

The OID is already exported by `github.com/google/certificate-transparency-go/x509` as `OIDExtensionCTPoison` but since that's an x509 fork it's annoying to use with stdlib x509 because of type mismatches (e.g. when comparing a cert `Extensions` value to the `OIDExtensionsCTPoison` oid).

Reply:
When in doubt I think it's fine to just start a new package, like `poison` so we'd have `poison.OID` :-)

Reply:
Hmm. Not in love with a package that only exists to have a single line/export but if that's the preference I'll do a follow-up PR to impl it across the codebase.

Reply:
We did something similar for the `canceled` package and it's worked out fine. I figure starting with microscopic packages and then consolidating if they proliferate too much is somewhat easier than starting with big packages and splitting them up.